### PR TITLE
refactor: make config package type-safe

### PR DESCRIPTION
**Pull request by elevatebart**
This PR is a part of #20521.

related to UNIFY-1216

It made sense to extract this out so I mitigate the risk.

### CODE REVIEW

File: packages/config/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  key = _.camelCase(key)
+
+  if (_.has(defaultValues, key)) {
+    return key
+  }
+
+  return
+}
+
+export { options }
```
Comment (lmiller1990) [inline packages/config/src/index.ts:122]: Can we export this at the top with the rest of the exports?

**Review by lmiller1990 — APPROVED** [review]
One nitpick but assuming no change in logic and just js->ts LGTM

File: packages/config/src/index.ts (hunk 1 above)
Comment (elevatebart) [reply packages/config/src/index.ts]: agreed

File: packages/config/__snapshots__/validation.spec.ts.js [hunk 2]
```diff
@@ -158,69 +189,38 @@ exports['array of non-strings'] = {
...
-    "fakeMode": 1
-  },
-  "type": "a positive number or null or an object with keys \"openMode\" and \"runMode\" with values of numbers or nulls"
-}
-
-exports['src/validation .isValidClientCertificatesSet returns error message for certs not passed as an array array 1'] = {
-  "key": "mockConfigKey",
-  "value": "1",
-  "type": "a positive number or null or an object with keys \"openMode\" and \"runMode\" with values of numbers or nulls"
+exports['not one of the strings error message'] = {
```
Comment (tgriesser) [inline packages/config/__snapshots__/validation.spec.ts.js:192]: Do we know why all of these changed? Can we copy over the old one to ensure nothing has changed here?

**Review by tgriesser — CHANGES_REQUESTED** [review]

File: packages/config/__snapshots__/validation.spec.ts.js (hunk 2 above)
Comment (elevatebart) [reply packages/config/__snapshots__/validation.spec.ts.js]: I re-ordered them in the same order as before

**Review by tgriesser — APPROVED** [review]
LGTM! Pending CI being green

**Comment by emilyrohrbough** [comment]
@elevatebart This didn't have two reviews before you merged this.

**Comment by elevatebart** [comment]
@emilyrohrbough I agree, and I apologize.
Lachlan's & Tim have approved and Zach had a look at it.

It was an honest mistake. 
Do you know how github dismisses approval?

**Comment by emilyrohrbough** [comment]
We have the repos setup to dismiss approvals if new commits are made after an approval.

**Pull request merged**